[FLINK-23559][tests] Randomize periodic materialisation interval in tests

rkhachatryan opened the pull request.

curcur commented on flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java:
```diff
@@ -129,6 +129,16 @@ private static void randomizeConfiguration(MiniCluster miniCluster, Configuratio
                         randomize(conf, StateChangelogOptions.ENABLE_STATE_CHANGE_LOG, true, false);
                 if (enabled) {
+                    randomize(
+                            conf,
+                            StateChangelogOptions.PERIODIC_MATERIALIZATION_INTERVAL,
+                            Duration.ofMillis(100),
+                            Duration.ofMillis(500),
+                            Duration.ofSeconds(1),
+                            Duration.ofSeconds(5),
+                            Duration.ofMinutes(10),
+                            Duration.ofSeconds(
+                                    Long.MAX_VALUE / 1000 /* max allowed by Duration.toMillis */));
```
This is a good idea.

But do we have ITTests that last more than 10 minutes?

If not I think the last two are equivalent saying that materialization is not played?

rkhachatryan replied:
I think you're right, 10 minutes is probably too much, I'll remove it.

curcur commented:
I've accepted the PR with some comments.

You can decide with your own judgement whether to keep the last two or not.

rkhachatryan commented:
Thanks for the review!
I removed the 10-minute option and rebased on the latest master (both required PRs are merged).
Will merge after passing the tests.

rkhachatryan commented:
Rebased to included the latests fixes, waiting for CI to merge.

rkhachatryan commented:
Private build passed except `e2e_2_ci_build` phase which passed in PR build, merging.
